### PR TITLE
[DO NOT REVIEW][SMOKE TEST] Skip prims shape in grad transform

### PR DESCRIPTION
**Pull request by jjsjann123**

### CODE REVIEW

File: thunder/core/transform_common.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def keep_or_swap(p):
         if not isinstance(p, NumberProxyInterface):
             return p
         if p.name in seen:
-            return p.value  # don't make it a duplicate
+            return None  # don't make it a duplicate
```
Comment (jjsjann123) [inline thunder/core/transform_common.py:126]: note for myself, I needed this to avoid an error in rematerialization pass. I'll work on a separate PR with a repro when I get that.

Comment (jjsjann123) [reply thunder/core/transform_common.py:126]: errrrr... this one doesn't seem to be working any more... I'm seeing the assert in rematerialization.py again.

Comment (jjsjann123) [reply thunder/core/transform_common.py:126]:
```
import thunder

def foo(a, b):
    return a + b

import torch

a = torch.randn(1, 32, 232, 232)
a.requires_grad_()
b = torch.randn(1, 1, 232, 232)
#b.requires_grad_()

jfoo = thunder.jit(foo, cache="symbolic values")

out = jfoo(a, b)
```

took me a while to get a repro here.
I think the issue is coming from saving for backward not properly identifying which gradient path isn't required. So after dce kicks in, the saved_for_backward is inconsistent... now I feel that I just missed a dce somewhere.

Comment (jjsjann123) [reply thunder/core/transform_common.py:126]: trying my luck with #1725

**Comment by jjsjann123** [comment]
closing inactive PRs.
